lib/spack/spack/bootstrap.py

```diff
@@ -196,6 +196,9 @@ def _raise_error(executable, exe_spec):
 
 
 def _bootstrap_config_scopes():
+    config_scopes = [
+        spack.config.InternalConfigScope('_builtin', spack.config.config_defaults)
+    ]
```

Comment:
I don't think this change is accounted, since `config_scopes` is reset just below:

https://github.com/spack/spack/blob/5383e0a35c8aa70f198b6f68c204788706e57f94/lib/spack/spack/bootstrap.py#L198-L203

Actually I am wondering why `spack style` is not signaling this typo :thinking:

Reply:
Do we have a bug to refer to, e.g. a missing property at some point during bootstrapping?

Reply:
Oh wow, I must have been sleeping. This was cherry-picked/rebased, that must have gone wrong.

Reply:
> Do we have a bug to refer to, e.g. a missing property at some point during bootstrapping?

Well, not really -- in the `-j` flag PR I assumed that `config:build_jobs` would always be set since it was set in _builtin, but apparently this was not the case in the tests, so I dropped that assumption. So we can also think about just closing this pr.

Reply:
Cherry picking sometimes is nasty. I cherry-picked the second commit in #22566 from a local branch not pushed to Github. Git was happy and I got some tests defined twice. Noticed just because of `spack style`.

Reply:
Although... `checksum = spack.config.get('config:checksum')` and `spack.config.get('config:verify_ssl')` are potentially used where `None` would act as `falsy`, so it might not hurt to set `_builtin` just in case the config files are deleted in some way?

Reply:
I'll check the options tomorrow and create an issue. I think `_builtin` should be part of the scopes.

Reply:
I think so too. I haven't run a bootstrap right now, but superficially it seems you could end up with unsafe installs of clingo & deps.